### PR TITLE
[Core] missing const in ModelPart

### PR DESCRIPTION
**Pull request by philbucher**
see title

- `GetSubModelPartNames` was not const
- Added the const version of `GetSubModelPart`
- Tests are added

### CODE REVIEW

**Review by loumalouomega — DISMISSED** [review]
Thanks

File: kratos/python/add_model_part_to_python.cpp [hunk 1]
```diff
@@ -850,7 +850,7 @@ void AddModelPartToPython(pybind11::module& m)
                 KRATOS_ERROR << "Setting geometries is not allowed! Trying to set value of ModelPart::Geometries."; })
         .def("CreateSubModelPart", &ModelPart::CreateSubModelPart, py::return_value_policy::reference_internal)
         .def("NumberOfSubModelParts", &ModelPart::NumberOfSubModelParts)
-        .def("GetSubModelPart", &ModelPart::GetSubModelPart, py::return_value_policy::reference_internal)
+        .def("GetSubModelPart", py::overload_cast<const std::string&>(&ModelPart::GetSubModelPart), py::return_value_policy::reference_internal) // non-const version
```
Comment (rubenzorrilla) [inline kratos/python/add_model_part_to_python.cpp:853]: :heart:

File: kratos/sources/model_part.cpp [hunk 2]
```diff
@@ -2049,6 +2049,30 @@ ModelPart& ModelPart::GetSubModelPart(std::string const& SubModelPartName)
...
+{
+    const auto delim_pos = SubModelPartName.find('.');
+    const std::string& sub_model_part_name = SubModelPartName.substr(0, delim_pos);
+
+    const auto i = mSubModelParts.find(sub_model_part_name);
+    if (i == mSubModelParts.end()) {
+        std::stringstream err_msg;
+        err_msg << "There is no sub model part with name \"" << SubModelPartName
+                << "\" in model part \"" << FullName() << "\"\n"
+                << "The the following sub model parts are available:";
```
Comment (rubenzorrilla) [inline kratos/sources/model_part.cpp]:
```suggestion
                << "The following sub model parts are available:";
```

File: kratos/sources/model_part.cpp [hunk 3]
```diff
@@ -2049,6 +2049,30 @@ ModelPart& ModelPart::GetSubModelPart(std::string const& SubModelPartName)
     }
 }
 
+const ModelPart& ModelPart::GetSubModelPart(std::string const& SubModelPartName) const
+{
+    const auto delim_pos = SubModelPartName.find('.');
+    const std::string& sub_model_part_name = SubModelPartName.substr(0, delim_pos);
```
Comment (rubenzorrilla) [inline kratos/sources/model_part.cpp]:
```suggestion
    const std::string& r_sub_model_part_name = SubModelPartName.substr(0, delim_pos);
```

File: kratos/sources/model_part.cpp [hunk 4]
```diff
@@ -2049,6 +2049,30 @@ ModelPart& ModelPart::GetSubModelPart(std::string const& SubModelPartName)
...
+
+    const auto i = mSubModelParts.find(sub_model_part_name);
```
Comment (rubenzorrilla) [inline kratos/sources/model_part.cpp]:
```suggestion
    const auto i = mSubModelParts.find(r_sub_model_part_name);
```

**Review by rubenzorrilla — COMMENTED** [review]
I missed this one many times. Thanks!

**Review by rubenzorrilla — APPROVED** [review]
Very good.

**Pull request merged**